pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractReplicator.java

```diff
@@ -163,16 +175,38 @@ public synchronized void startProducer() {
 
     }
 
-    protected void checkTopicActiveAndRetryStartProducer() {
-        isLocalTopicActive().thenAccept(isTopicActive -> {
-            if (isTopicActive) {
-                startProducer();
+    protected void scheduleCheckTopicActiveAndStartProducer(final long waitTimeMs) {
+        brokerService.executor().schedule(() -> {
+            if (state == State.Terminated) {
```

Comment:
```suggestion
             if (state == State.Terminating || state == State.Terminated) {
```

And it's better to add an info level log here so that we can know why the producer creation is skipped.

Reply:
And it looks like we also have 4 more case below to skip the producer creation. Is it better to have a log for each of them? We can have a method to print the log with the skipped reason.

Reply:
> And it's better to add an info level log here so that we can know why the producer creation is skipped.

Done

> And it looks like we also have 4 more case below to skip the producer creation. Is it better to have a log for each of them? We can have a method to print the log with the skipped reason.

Done